### PR TITLE
Use person tested obs to display screening business status in family profile

### LINKED ISSUE #658: RVL-63 - Historical Form Change: 18: RACD business status = tested or not tested
**NEW DESCRIPTION**
    Login to mobile client.
    Select OA and Plan.
    select structure to complete family registration.
    Go to task module.
    Record Screening.
**Current behavior:**
After data capture, the task status changes to COMPETE whether or not the member was tested.

**Desired Behavior:**
After data capture, the task status should be in green with text “Tested” if family member was tested or “Not Tested” if family member was not tested.

### PR DESCRIPTION
**Pull request by Rkareko**
Resolves #658

### CODE REVIEW

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-reveal/src/main/java/org/smartregister/reveal/viewholder/StructureTaskViewHolder.java  4
- opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/StructureTasksInteractor.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5165710&bid=16959365)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/StructureTasksInteractor.java  1
- opensrp-reveal/src/main/java/org/smartregister/reveal/viewholder/StructureTaskViewHolder.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5165710&bid=16959365)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/StructureTasksInteractor.java  1
- opensrp-reveal/src/main/java/org/smartregister/reveal/viewholder/StructureTaskViewHolder.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5165710&bid=16959365)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/StructureTasksInteractor.java  1
- opensrp-reveal/src/main/java/org/smartregister/reveal/viewholder/StructureTaskViewHolder.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5165710&bid=16959365)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-reveal/src/main/java/org/smartregister/reveal/viewholder/StructureTaskViewHolder.java  5
- opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/StructureTasksInteractor.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5165710&bid=16959365)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-reveal/src/main/java/org/smartregister/reveal/viewholder/StructureTaskViewHolder.java  5
- opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/StructureTasksInteractor.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5165710&bid=16959365)

File: opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/StructureTasksInteractor.java [hunk 1]
```diff
@@ -193,6 +195,11 @@ private void populateEventsPerTask(List<StructureTaskDetails> tasks) {
                     lastEventDate.bindString(1, task.getTaskId());
                     eventTask.setLastEventDate(lastEventDate.simpleQueryForString());
                     task.setLastEdited(DateUtil.parseDate(eventTask.getLastEventDate()));
+
+                }
+
+                if (Intervention.BLOOD_SCREENING.equals(task.getTaskCode())){
+                    setPersonTested(task,eventTask.getEventsPerTask() > 1);
```
Comment (githengi) [inline opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/StructureTasksInteractor.java]: shouldn't this be `>0`

File: opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/StructureTasksInteractor.java [hunk 2]
```diff
@@ -210,6 +217,21 @@ private void populateEventsPerTask(List<StructureTaskDetails> tasks) {
 
     }
 
+    private void setPersonTested(StructureTaskDetails task, boolean isEdit) {
+        if (isEdit){
+            String personTestWithEditsSql = "select person_tested from event_task where id in " +
+                    "(select formSubmissionId from event where baseEntityId = ? and eventType = ? " +
+                    "order by updatedAt desc limit 1)";
+            SQLiteStatement personTestWithEdits = database.compileStatement(personTestWithEditsSql);
```
Comment (githengi) [inline opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/StructureTasksInteractor.java]: statement is not closed

File: opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/StructureTasksInteractor.java [hunk 3]
```diff
@@ -210,6 +217,21 @@ private void populateEventsPerTask(List<StructureTaskDetails> tasks) {
...
+            personTestWithEdits.bindString(1, task.getTaskEntity());
+            personTestWithEdits.bindString(2, BLOOD_SCREENING_EVENT);
+            task.setPersonTested(personTestWithEdits.simpleQueryForString());
+        } else {
+            SQLiteStatement personTested = database.compileStatement("SELECT person_tested FROM event_task WHERE task_id = ?");
```
Comment (githengi) [inline opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/StructureTasksInteractor.java]: statement is not closed

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/StructureTasksInteractor.java  1
- opensrp-reveal/src/main/java/org/smartregister/reveal/viewholder/StructureTaskViewHolder.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5165710&bid=16959365)

**Pull request merged**